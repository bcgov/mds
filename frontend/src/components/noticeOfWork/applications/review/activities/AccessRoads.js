import React from "react";
import { PropTypes } from "prop-types";
import { Field } from "redux-form";
import { Row, Col, Table } from "antd";
import * as Strings from "@/constants/strings";
import RenderField from "@/components/common/RenderField";
import RenderAutoSizeField from "@/components/common/RenderAutoSizeField";
import RenderRadioButtons from "@/components/common/RenderRadioButtons";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
};

export const AccessRoads = (props) => {
  const columns = [
    {
      title: "Access Type",
      dataIndex: "type",
      key: "type",
      render: (text) => <div title="Access Type">{text}</div>,
    },
    {
      title: "Length(km)",
      dataIndex: "length",
      key: "length",
      render: (text) => <div title="Length(km)">{text}</div>,
    },
    {
      title: "Disturbed Area (ha)",
      dataIndex: "disturbedArea",
      key: "disturbedArea",
      render: (text) => <div title="Disturbed Area (ha)">{text}</div>,
    },
    {
      title: "Merchantable timber volume (m3)",
      dataIndex: "timberVolume",
      key: "timberVolume",
      render: (text) => <div title="Merchantable timber volume (m3)">{text}</div>,
    },
  ];

  const transformData = (activities) =>
    activities.map((activity) => ({
      type: activity.activity_type_description || Strings.EMPTY_FIELD,
      length: activity.length || Strings.EMPTY_FIELD,
      disturbedArea: activity.disturbed_area || Strings.EMPTY_FIELD,
      timberVolume: activity.timber_volume || Strings.EMPTY_FIELD,
    }));

  return (
    <div>
      <Table
        align="left"
        pagination={false}
        columns={columns}
        dataSource={transformData(props.initialValues.details ? props.initialValues.details : [])}
        locale={{
          emptyText: "No data",
        }}
      />
      <br />
      <h4>Bridges, Culverts, and Crossings</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Are you proposing any bridges, culverts, and crossings?**
          </div>
          <Field id="" name="" component={RenderRadioButtons} disabled={props.isViewMode} />
          <Field id="" name="" component={RenderField} disabled={props.isViewMode} />
        </Col>
      </Row>
      <br />
      <h4>Reclamation Program</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Proposed reclamation and timing for this specific activity
          </div>
          <Field
            id="reclamation_description"
            name="reclamation_description"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">
            Estimated Cost of reclamation activities described above
          </div>
          <Field
            id="reclamation_cost"
            name="reclamation_cost"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
    </div>
  );
};

AccessRoads.propTypes = propTypes;

export default AccessRoads;
