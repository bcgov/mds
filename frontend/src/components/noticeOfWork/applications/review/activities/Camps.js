import React from "react";
import { PropTypes } from "prop-types";
import { Field } from "redux-form";
import { Row, Col } from "antd";
import RenderField from "@/components/common/RenderField";
import RenderAutoSizeField from "@/components/common/RenderAutoSizeField";
import RenderRadioButtons from "@/components/common/RenderRadioButtons";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
};

export const Camps = (props) => {
  return (
    <div>
      {/* <br />
      <h4>Camps</h4>

      <br />
      <h4>Buildings</h4>

      <br />
      <h4>Staging Area</h4> */}
      <br />
      <h4>Fuel</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Do you propose to store fuel?**</div>
          <Field
            id="has_fuel_stored"
            name="has_fuel_stored"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">Storage Method**</div>
          <Row gutter={16}>
            <Col md={12} sm={24}>
              <Field
                label="Bulk"
                id="has_fuel_stored_in_bulk"
                name="has_fuel_stored_in_bulk"
                component={RenderRadioButtons}
                disabled={props.isViewMode}
              />
            </Col>
            <Col md={12} sm={24}>
              <Field
                label="Barrell"
                id="has_fuel_stored_in_barrels"
                name="has_fuel_stored_in_barrels"
                component={RenderRadioButtons}
                disabled={props.isViewMode}
              />
            </Col>
          </Row>
        </Col>
      </Row>
      <br />
      <h4>Reclamation Program</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Describe the proposed reclamation and timing for this specific activity**
          </div>
          <Field
            id="reclamation_description"
            name="reclamation_description"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">
            Estimated Cost of reclamation activities described above**
          </div>
          <Field
            id="reclamation_cost"
            name="reclamation_cost"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
    </div>
  );
};

Camps.propTypes = propTypes;

export default Camps;
