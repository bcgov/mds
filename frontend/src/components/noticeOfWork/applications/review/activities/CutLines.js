import React from "react";
import { PropTypes } from "prop-types";
import { Field } from "redux-form";
import { Row, Col } from "antd";
import RenderField from "@/components/common/RenderField";
import RenderAutoSizeField from "@/components/common/RenderAutoSizeField";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
};

export const CutLines = (props) => {
  return (
    <div>
      {/* <br />
      <h4>Exploration Grid</h4> */}
      <br />
      <h4>Reclamation Program</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Describe the proposed reclamation and timing for this specific activity**
          </div>
          <Field
            id="reclamation_description"
            name="reclamation_description"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">
            Estimated Cost of reclamation activities described above**
          </div>
          <Field
            id="reclamation_cost"
            name="reclamation_cost"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
    </div>
  );
};

CutLines.propTypes = propTypes;

export default CutLines;
