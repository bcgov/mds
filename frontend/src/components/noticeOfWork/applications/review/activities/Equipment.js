import React from "react";
import { PropTypes } from "prop-types";
import { Table } from "antd";
import * as Strings from "@/constants/strings";

const propTypes = {
  initialValues: PropTypes.objectOf(PropTypes.string).isRequired,
};

export const Equipment = (props) => {
  const columns = [
    {
      title: "Quantity",
      dataIndex: "quantity",
      key: "quantity",
      render: (text) => <div title="Quantity">{text}</div>,
    },
    {
      title: "Description",
      dataIndex: "description",
      key: "description",
      render: (text) => <div title="Description">{text}</div>,
    },
    {
      title: "Capacity",
      dataIndex: "capacity",
      key: "capacity",
      render: (text) => <div title="Capacity">{text}</div>,
    },
  ];

  const transformData = (equipmentList) =>
    equipmentList.map((equipment) => ({
      quantity: equipment.quantity || Strings.EMPTY_FIELD,
      description: equipment.description || Strings.EMPTY_FIELD,
      capacity: equipment.capacity || Strings.EMPTY_FIELD,
    }));

  return (
    <div>
      <h3>Equipment</h3>
      <Table
        align="left"
        pagination={false}
        columns={columns}
        dataSource={transformData(props.initialValues.equipment)}
        locale={{
          emptyText: "No equipment related to this activity",
        }}
      />
    </div>
  );
};

Equipment.propTypes = propTypes;

export default Equipment;
