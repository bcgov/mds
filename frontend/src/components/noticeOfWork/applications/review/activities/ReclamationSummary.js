import React from "react";
import { PropTypes } from "prop-types";
import { Table } from "antd";

const propTypes = {
  summary: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
};

export const ReclamationSummary = (props) => {
  const columns = [
    {
      title: "Activity",
      dataIndex: "label",
      key: "label",
      render: (text) => <div title="Activity">{text}</div>,
    },
    {
      title: "Total Effected Area (ha)",
      dataIndex: "total",
      key: "total",
      render: (text) => <div title="Total Effected Area (ha)">{text}</div>,
    },
    {
      title: "Estimated Cost of Reclamation",
      dataIndex: "cost",
      key: "cost",
      render: (text) => <div title="Estimated Cost of Reclamation">{text}</div>,
    },
  ];

  return (
    <div>
      <h3>Reclamation Summary</h3>
      <Table
        align="left"
        pagination={false}
        columns={columns}
        dataSource={props.summary}
        locale={{
          emptyText: "No data",
        }}
      />
    </div>
  );
};

ReclamationSummary.propTypes = propTypes;

export default ReclamationSummary;
