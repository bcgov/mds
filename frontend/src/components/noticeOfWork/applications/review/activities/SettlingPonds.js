import React from "react";
import { PropTypes } from "prop-types";
import { Field } from "redux-form";
import { Row, Col, Table } from "antd";
import * as Strings from "@/constants/strings";
import RenderField from "@/components/common/RenderField";
import RenderAutoSizeField from "@/components/common/RenderAutoSizeField";
import RenderRadioButtons from "@/components/common/RenderRadioButtons";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.string).isRequired,
};

export const SettlingPonds = (props) => {
  const columns = [
    {
      title: "Pond ID",
      dataIndex: "id",
      key: "id",
      render: (text) => <div title="Pond ID">{text}</div>,
    },
    {
      title: "Width(m)",
      dataIndex: "width",
      key: "width",
      render: (text) => <div title="Width(m)">{text}</div>,
    },
    {
      title: "Length(km)",
      dataIndex: "length",
      key: "length",
      render: (text) => <div title="Length(km)">{text}</div>,
    },
    {
      title: "Depth(m)",
      dataIndex: "depth",
      key: "depth",
      render: (text) => <div title="Depth(m)">{text}</div>,
    },
    {
      title: "Disturbed Area (ha)",
      dataIndex: "disturbedArea",
      key: "disturbedArea",
      render: (text) => <div title="Disturbed Area (ha)">{text}</div>,
    },
    {
      title: "Merchantable timber volume (m3)",
      dataIndex: "timberVolume",
      key: "timberVolume",
      render: (text) => <div title="Merchantable timber volume (m3)">{text}</div>,
    },
  ];

  const transformData = (activities) =>
    activities.map((activity) => ({
      id: activity.activity_type_description || Strings.EMPTY_FIELD,
      width: activity.width || Strings.EMPTY_FIELD,
      depth: activity.depth || Strings.EMPTY_FIELD,
      length: activity.length || Strings.EMPTY_FIELD,
      disturbedArea: activity.disturbed_area || Strings.EMPTY_FIELD,
      timberVolume: activity.timber_volume || Strings.EMPTY_FIELD,
    }));

  return (
    <div>
      <Table
        align="left"
        pagination={false}
        columns={columns}
        dataSource={transformData(props.initialValues.details ? props.initialValues.details : [])}
        locale={{
          emptyText: "No data",
        }}
      />
      <br />
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Disposal of fines from clean out (i.e. use as a subsoil material)**
          </div>
          <Field id="" name="" component={RenderAutoSizeField} disabled={props.isViewMode} />
        </Col>
      </Row>
      <Row gutter={16}>
        <div className="field-title">Water from Ponds will be</div>
        <Col md={8} sm={24}>
          <Field
            label="Recycled"
            id="is_ponds_recycled"
            name="is_ponds_recycled"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={8} sm={24}>
          <Field
            label="Exfiltrated to Ground"
            id="is_ponds_exfiltrated"
            name="is_ponds_exfiltrated"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={8} sm={24}>
          <Field
            label="Discharged to Environment"
            id="is_ponds_discharged"
            name="is_ponds_discharged"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <br />
      <h4>Reclamation Program</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Proposed reclamation and timing for this specific activity**
          </div>
          <Field
            id="reclamation_description"
            name="reclamation_description"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">
            Estimated Cost of reclamation activities described above**
          </div>
          <Field
            id="reclamation_cost"
            name="reclamation_cost"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
    </div>
  );
};

SettlingPonds.propTypes = propTypes;

export default SettlingPonds;
