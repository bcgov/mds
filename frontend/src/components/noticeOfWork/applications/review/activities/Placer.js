import React from "react";
import { PropTypes } from "prop-types";
import { Field } from "redux-form";
import { Row, Col, Table } from "antd";
import * as Strings from "@/constants/strings";
import RenderRadioButtons from "@/components/common/RenderRadioButtons";
import RenderAutoSizeField from "@/components/common/RenderAutoSizeField";
import RenderField from "@/components/common/RenderField";
import Equipment from "@/components/noticeOfWork/applications/review/activities/Equipment";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.string).isRequired,
};

export const Placer = (props) => {
  const columns = [
    {
      title: "Activity",
      dataIndex: "type",
      key: "type",
      render: (text) => <div title="Activity">{text}</div>,
    },
    {
      title: "Quantity",
      dataIndex: "quantity",
      key: "quantity",
      render: (text) => <div title="Quantity">{text}</div>,
    },
    {
      title: "Width(m)",
      dataIndex: "width",
      key: "width",
      render: (text) => <div title="Width(m)">{text}</div>,
    },
    {
      title: "Length(km)",
      dataIndex: "length",
      key: "length",
      render: (text) => <div title="Length(km)">{text}</div>,
    },
    {
      title: "Disturbed Area (ha)",
      dataIndex: "disturbedArea",
      key: "disturbedArea",
      render: (text) => <div title="Disturbed Area (ha)">{text}</div>,
    },
    {
      title: "Merchantable timber volume (m3)",
      dataIndex: "timberVolume",
      key: "timberVolume",
      render: (text) => <div title="Merchantable timber volume (m3)">{text}</div>,
    },
  ];

  const transformData = (activities) =>
    activities.map((activity) => ({
      type: activity.activity_type_description || Strings.EMPTY_FIELD,
      width: activity.width || Strings.EMPTY_FIELD,
      quantity: activity.quantity || Strings.EMPTY_FIELD,
      length: activity.length || Strings.EMPTY_FIELD,
      disturbedArea: activity.disturbed_area || Strings.EMPTY_FIELD,
      timberVolume: activity.timber_volume || Strings.EMPTY_FIELD,
    }));

  return (
    <div>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Is this an application for Underground Placer Operations?
          </div>
          <Field
            id="is_underground"
            name="is_underground"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">Is this an application for Hand Operations?</div>
          <Field
            id="is_hand_operation"
            name="is_hand_operation"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <br />
      <Table
        align="left"
        pagination={false}
        columns={columns}
        dataSource={transformData(props.initialValues.details ? props.initialValues.details : [])}
        locale={{
          emptyText: "No data",
        }}
      />
      <br />
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Proposed Production**</div>
          <Field id="" name="" component={RenderField} disabled={props.isViewMode} />
        </Col>
      </Row>
      <br />
      {props.initialValues.equipment && <Equipment equipment={props.initialValues.equipment} />}

      <h4>Reclamation Program</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Total area of planned reclamation this year</div>
          <Field
            id="total_disturbed_area"
            name="total_disturbed_area"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">
            Proposed reclamation and timing for this specific activity
          </div>
          <Field
            id="reclamation_description"
            name="reclamation_description"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">
            Estimated Cost of reclamation activities described above
          </div>
          <Field
            id="reclamation_cost"
            name="reclamation_cost"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
    </div>
  );
};

Placer.propTypes = propTypes;

export default Placer;
