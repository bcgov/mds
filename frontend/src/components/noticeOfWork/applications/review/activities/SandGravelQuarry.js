import React from "react";
import { PropTypes } from "prop-types";
import { Field } from "redux-form";
import { Row, Col, Table } from "antd";
import * as Strings from "@/constants/strings";
import RenderField from "@/components/common/RenderField";
import RenderAutoSizeField from "@/components/common/RenderAutoSizeField";
import RenderRadioButtons from "@/components/common/RenderRadioButtons";
import Equipment from "@/components/noticeOfWork/applications/review/activities/Equipment";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
};

export const SandGravelQuarry = (props) => {
  const columns = [
    {
      title: "Activity",
      dataIndex: "type",
      key: "type",
      render: (text) => <div title="Activity">{text}</div>,
    },
    {
      title: "Length(km)",
      dataIndex: "length",
      key: "length",
      render: (text) => <div title="Length(km)">{text}</div>,
    },
    {
      title: "Disturbed Area (ha)",
      dataIndex: "disturbedArea",
      key: "disturbedArea",
      render: (text) => <div title="Disturbed Area (ha)">{text}</div>,
    },
    {
      title: "Merchantable timber volume (m3)",
      dataIndex: "timberVolume",
      key: "timberVolume",
      render: (text) => <div title="Merchantable timber volume (m3)">{text}</div>,
    },
  ];

  const transformData = (activities) =>
    activities.map((activity) => ({
      type: activity.activity_type_description || Strings.EMPTY_FIELD,
      length: activity.length || Strings.EMPTY_FIELD,
      disturbedArea: activity.disturbed_area || Strings.EMPTY_FIELD,
      timberVolume: activity.timber_volume || Strings.EMPTY_FIELD,
    }));

  return (
    <div>
      <br />
      <h4>Soil Conservation</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Average Depth Overburden(m)</div>
          <Field
            id="average_overburden_depth"
            name="average_overburden_depth"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">
            Measures to stabilize soil overburden stockpiles and control noxious weeds
          </div>
          <Field
            id="stability_measures_description"
            name="stability_measures_description"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Average Depth of top soil(m)</div>
          <Field
            id="average_top_soil_depth"
            name="average_top_soil_depth"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>

      <br />
      <h4>Land Use</h4>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Is this site within the Agricultural Land Reserve?</div>
          <Field
            id="is_agricultural_land_reserve"
            name="is_agricultural_land_reserve"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">Current land use zoning for the site </div>
          <Field
            id="land_use_zoning"
            name="land_use_zoning"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title--light">Permit Application Number**</div>
          <Field id="" name="" component={RenderField} disabled={props.isViewMode} />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">Proposed land use</div>
          <Field
            id="proposed_land_use"
            name="proposed_land_use"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Does the local government have a soil removal bylaw?</div>
          <Field
            id="has_local_soil_removal_bylaw"
            name="has_local_soil_removal_bylaw"
            component={RenderRadioButtons}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">
            Estimate total mineable reserves over the life of the mine(m3)
          </div>
          <Field
            id="total_mineable_reserves"
            name="total_mineable_reserves"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <Row gutter={16}>
        <Col md={12} sm={24}>
          <div className="field-title">Official community plan for the site</div>
          <Field
            id="community_plan"
            name="community_plan"
            component={RenderField}
            disabled={props.isViewMode}
          />
        </Col>
        <Col md={12} sm={24}>
          <div className="field-title">Estimate annual extraction from site (tonnes/year)</div>
          <Field
            id="total_annual_extraction"
            name="total_annual_extraction"
            component={RenderAutoSizeField}
            disabled={props.isViewMode}
          />
        </Col>
      </Row>
      <br />
      <Table
        align="left"
        pagination={false}
        columns={columns}
        dataSource={transformData(props.initialValues.details ? props.initialValues.details : [])}
        locale={{
          emptyText: "No data",
        }}
      />
      <br />
      {props.initialValues.equipment && <Equipment equipment={props.initialValues.equipment} />}
    </div>
  );
};

SandGravelQuarry.propTypes = propTypes;

export default SandGravelQuarry;
