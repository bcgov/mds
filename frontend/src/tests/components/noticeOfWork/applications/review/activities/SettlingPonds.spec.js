import React from "react";
import { shallow } from "enzyme";
import { SettlingPonds } from "@/components/noticeOfWork/applications/review/activities/SettlingPonds";
import * as NOW_MOCK from "@/tests/mocks/noticeOfWorkMocks";

const reducerProps = {};

const setupReducerProps = () => {
  reducerProps.isViewMode = true;
  reducerProps.initialValues = NOW_MOCK.IMPORTED_NOTICE_OF_WORK.settling_pond;
};

beforeEach(() => {
  setupReducerProps();
});

describe("SettlingPonds", () => {
  it("renders properly", () => {
    const component = shallow(<SettlingPonds {...reducerProps} />);
    expect(component).toMatchSnapshot();
  });
});
