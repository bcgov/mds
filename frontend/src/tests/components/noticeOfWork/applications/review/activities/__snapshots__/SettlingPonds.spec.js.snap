// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettlingPonds renders properly 1`] = `
<div>
  <Row
    gutter={16}
  >
    <Col
      md={12}
      sm={24}
    >
      <div
        className="field-title"
      >
        Describe the waste water treatment facility (settling pond design, recycling, distance from creek, etc.)**
      </div>
      <Field
        component={[Function]}
        disabled={true}
        id=""
        name=""
      />
    </Col>
  </Row>
  <br />
  <withStore(Table)
    align="left"
    columns={
      Array [
        Object {
          "dataIndex": "id",
          "key": "id",
          "render": [Function],
          "title": "Pond ID",
        },
        Object {
          "dataIndex": "width",
          "key": "width",
          "render": [Function],
          "title": "Width(m)",
        },
        Object {
          "dataIndex": "length",
          "key": "length",
          "render": [Function],
          "title": "Length(km)",
        },
        Object {
          "dataIndex": "depth",
          "key": "depth",
          "render": [Function],
          "title": "Depth(m)",
        },
        Object {
          "dataIndex": "disturbedArea",
          "key": "disturbedArea",
          "render": [Function],
          "title": "Disturbed Area (ha)",
        },
        Object {
          "dataIndex": "timberVolume",
          "key": "timberVolume",
          "render": [Function],
          "title": "Merchantable timber volume (m3)",
        },
      ]
    }
    dataSource={Array []}
    locale={
      Object {
        "emptyText": "No data",
      }
    }
    pagination={false}
  />
  <br />
  <Row
    gutter={16}
  >
    <Col
      md={12}
      sm={24}
    >
      <div
        className="field-title"
      >
        Disposal of fines from clean out (i.e. use as a subsoil material)**
      </div>
      <Field
        component={[Function]}
        disabled={true}
        id=""
        name=""
      />
    </Col>
  </Row>
  <Row
    gutter={16}
  >
    <div
      className="field-title"
    >
      Water from Ponds will be
    </div>
    <Col
      md={8}
      sm={24}
    >
      <Field
        component={[Function]}
        disabled={true}
        id="is_ponds_recycled"
        label="Recycled"
        name="is_ponds_recycled"
      />
    </Col>
    <Col
      md={8}
      sm={24}
    >
      <Field
        component={[Function]}
        disabled={true}
        id="is_ponds_exfiltrated"
        label="Exfiltrated to Ground"
        name="is_ponds_exfiltrated"
      />
    </Col>
    <Col
      md={8}
      sm={24}
    >
      <Field
        component={[Function]}
        disabled={true}
        id="is_ponds_discharged"
        label="Discharged to Environment"
        name="is_ponds_discharged"
      />
    </Col>
  </Row>
  <br />
  <h4>
    Reclamation Program
  </h4>
  <Row
    gutter={16}
  >
    <Col
      md={12}
      sm={24}
    >
      <div
        className="field-title"
      >
        Proposed reclamation and timing for this specific activity**
      </div>
      <Field
        component={[Function]}
        disabled={true}
        id="reclamation_description"
        name="reclamation_description"
      />
    </Col>
    <Col
      md={12}
      sm={24}
    >
      <div
        className="field-title"
      >
        Estimated Cost of reclamation activities described above**
      </div>
      <Field
        component={[Function]}
        disabled={true}
        id="reclamation_cost"
        name="reclamation_cost"
      />
    </Col>
  </Row>
</div>
`;
